x/ibc/core/04-channel/handler.go

```diff
@@ -18,20 +18,24 @@ func HandleMsgChannelOpenInit(ctx sdk.Context, k keeper.Keeper, portCap *capabil
 		return nil, "", nil, sdkerrors.Wrap(err, "channel handshake open init failed")
 	}
 
-	ctx.EventManager().EmitEvents(sdk.Events{
-		sdk.NewEvent(
-			types.EventTypeChannelOpenInit,
-			sdk.NewAttribute(types.AttributeKeyPortID, msg.PortId),
-			sdk.NewAttribute(types.AttributeKeyChannelID, channelID),
-			sdk.NewAttribute(types.AttributeCounterpartyPortID, msg.Channel.Counterparty.PortId),
-			sdk.NewAttribute(types.AttributeCounterpartyChannelID, msg.Channel.Counterparty.ChannelId),
-			sdk.NewAttribute(types.AttributeKeyConnectionID, msg.Channel.ConnectionHops[0]),
-		),
+	if err := ctx.EventManager().EmitTypedEvent(
+		&types.EventChannelOpenInit{
+			PortId:                msg.PortId,
+			ChannelId:             channelID,
+			CounterpartyPortId:    msg.Channel.Counterparty.PortId,
+			CounterpartyChannelId: msg.Channel.Counterparty.ChannelId,
+			ConnectionId:          msg.Channel.ConnectionHops[0],
+		},
+	); err != nil {
+		return nil, "", nil, err
+	}
+
+	ctx.EventManager().EmitEvent(
```

Comment:
I think we can remove these now. I think they were there to help with parsing events, but this pr should make that unnecessary right?